src/hotspot/share/utilities/vmError.cpp

```diff
@@ -1501,27 +1500,53 @@ void VMError::report_and_die(int id, const char* message, const char* detail_fmt
         // Watcherthread is about to call os::die. Lets just wait.
         os::infinite_sleep();
       } else {
-        // Crash or assert during error reporting. Lets continue reporting with the next step.
-        stringStream ss(buffer, sizeof(buffer));
+        // A secondary error happened. Print brief information, but take care, since crashing
+        // here would just recurse endlessly.
+        // Any information (signal, context, siginfo etc) printed here should use the function
+        // arguments, not the information stored in *this, since those describe the primary crash.
+        static char tmp[256]; // cannot use global scratch buffer
```

Comment:
I'm not sure how this would happen: we only end up here if we are the original thread that triggered the first fault (_first_error_tid == me). Otherwise, we would end up in `os::infinite_sleep()` at lines 1455ff. If a different thread crashes inside signal handling, between signal handler start and infinite_sleep, it will just keep crashing endlessly and run out of stack.